app/models/health/engagement_calculator.rb

```diff
@@ -0,0 +1,85 @@
+###
+# Copyright 2016 - 2022 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+module Health
+  class EngagementCalculator
+    include Rails.application.routes.url_helpers
+
+    def initialize(patient)
+      @patient = patient
+      @client = patient.client
+      @valid_after = patient.contributing_enrollment_start_date - 1.year
+    end
+
+    # TODO 10/4/22 Participation form is no longer used, left here for now
+    def participation_form_status
+      signed_participation_form = @patient.participation_forms.recent.first
+      return [:no_signed_form, 'Participation Form', new_client_health_participation_form_path(@client), true, nil] if signed_participation_form.blank?
+      return [:too_old, 'Participation Form', new_client_health_participation_form_path(@client), true, "Last signed on #{signed_participation_form.signature_on}"] if signed_participation_form.signature_on < @valid_after
+
+      [:valid, 'Participation Form', edit_client_health_participation_form_path(@client, signed_participation_form), true, nil]
+    end
+
+    def release_form_status
+      signed_release_form = @patient.release_forms.recent.first
+      return [:no_signed_form, 'Participation and Release of Information', new_client_health_release_form_path(@client), true, nil] if signed_release_form.blank?
+      return [:too_old, 'Participation and Release of Information', new_client_health_release_form_path(@client), true, "Last signed on #{signed_release_form.signature_on}"] if signed_release_form.signature_on < @valid_after
+      return [:expired, 'Participation and Release of Information', new_client_health_release_form_path(@client), true, "Expired on #{signed_release_form.signature_on + 2.years}"] if signed_release_form.signature_on + 2.years <= Date.current
+
+      [:valid, 'Participation and Release of Information', edit_client_health_release_form_path(@client, signed_release_form), true, nil]
```

Comment:
I'm probably missing it, but doesn't this need to check the participation signature somewhere in there?

Reply:
As you noticed below, they can't sign a release without having a participation signature, so this shouldn't be a problem.